src/libraries/System.Collections.Immutable/src/Resources/Strings.resx

```diff
@@ -105,4 +105,7 @@
   <data name="ArgumentOutOfRange_NeedNonNegNum" xml:space="preserve">
     <value>Non-negative number required.</value>
   </data>
+  <data name="InvalidOperation_IncompatibleComparer" xml:space="preserve">
+    <value>The collection's comparer does not support the requested operation.</value>
```

Comment:
Could this exception message be more specific, e.g. by explaining that the alternate key type is not compatible? It could also explicitly state which types of keys are current compatible for frozen collections.